R/expect-that.R

```diff
@@ -50,11 +50,10 @@ expect_that <- function(object, condition, info = NULL, label = NULL) {
 #' test_that("this test fails", fail())
 #' test_that("this test succeeds", succeed())
 #' }
-fail <- function(message = "Failure has been forced", info = NULL) {
-  expect(FALSE, message, info = info)
+fail <- function(message = "Failure has been forced", info = NULL, trace = NULL) {
```

Comment:
Are you planning to make this a common pattern? We probably want to avoid creating traces ahead of time and discarding them if the expectation succeeds because that will bring performance down.

Reply:
I think the hint about passing around bottom envs resolves this — I can just pass around the bottom env, and then only create the traceback when needed. Not sure how I forgot about that when I was using it for the top env 😆